### ISSUE TITLE
local htmltest runs no longer pass in a post-managed-docs world

### ISSUE THREAD

**Issue by rmloveland** [cockroachdb/docs#4093]
Sorry if there is an issue filed against this behavior already, a search for "is:open htmltest" didn't find it!

tl;dr: `htmltest` no longer passes when run locally, so we're flying blind until we push to Github and see the green check (I think?  Happy to be wrong about this, maybe I'm doing something wrong locally).

Relates to #3301, #3833.

Full version:

If you run `make standard` and your build output directory contains `_site/docs`, local runs of `htmltest -s` return the following:

    htmltest -s
    Skipping the checking of external links.
    htmltest started at 03:32:28 on _site
    ========================================================================
    running in concurrent mode, this is experimental
    docs/releases/v2.1.0.html
      target does not exist --- docs/releases/v2.1.0.html --> ../managed/v2.1/
      target does not exist --- docs/releases/v2.1.0.html --> ../managed/v2.1/managed-sign-up-for-a-cluster.html
    docs/v2.1/frequently-asked-questions.html
      target does not exist --- docs/v2.1/frequently-asked-questions.html --> ../managed/v2.1/
    docs/v2.2/frequently-asked-questions.html
      target does not exist --- docs/v2.2/frequently-asked-questions.html --> ../managed/v2.1/
    ========================================================================
    âââ failed in 2.196259095s
    4 errors in 1962 documents

If you then run `make managed` and your build output directory contains both `_site/docs` and `_site/managed`, local runs of `htmltest -s` return a giant list of errors that I will attach below.

[foo.txt](https://github.com/cockroachdb/docs/files/2605750/foo.txt)

**Comment by rmloveland** [comment]
It occurs to me that we could get this case to pass (`make standard`) by adding some logic to create a few dummy pages in the `_site/managed` folder, but that's probably not the Right Thing (?).

**Comment by otan** [comment]
took a brief look at this, what does this pass in on atm?
it does seem like genuine issues are being hit.

**Comment by otan** [comment]
also i think the two related issues are unrelated and separate!

**Comment by rmloveland** [comment]
> took a brief look at this, what does this pass in on atm?

@otan sorry, I don't understand exactly what you are asking.  Do you mean what args are getting passed to `htmltest` when it's running locally?

At least in my usage, it's getting run basically as follows:

```
$ cd /path/to/cockroachdb/docs
$ make standard-build # generates _site/docs directory with HTML files
$ htmltest -s
# broken link output here
```

**Comment by otan** [comment]
It seems like they were genuinely broken links. I'm asking whether this passes in CI (and I could be misreading the output!)

**Comment by rmloveland** [comment]
Oliver Tan <notifications@github.com> writes:

> It seems like they were genuinely broken links. I'm asking whether
> this passes in CI (and I could be misreading the output!)

AFAICT it does pass CI.   That's the core of the issue IMO - output of
`htmltest` looks totally broken locally and yet the CI still passes.

I think the reason for the breakage is:

- we have two build outputs: 'standard' and 'cockroachcloud', which go
  to different directories

- there are links from the 'standard' docs to the 'cockroachcloud' docs,
  but when you do a 'standard' build locally, the 'cockroachcloud' build
  output it's pointing to does not exist

**Comment by rmloveland** [comment]
PS you are not misreading the output - the links are genuinely broken locally (~intentionally) due to the design described above in "reason for the breakage", which it would be great to address.
 
I'm also happy to discuss in-person / via video or something if that would be easier!

**Comment by otan** [comment]
Okay! So I can get this working locally by running the following:

_config_prod_cockroachdb.yml:
```
url: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com
baseurl: //cockroachcloud
destination: _site//cockroachcloud
```

_config_prod_cockroachcloud.yml:
```
url: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com
baseurl: /
destination: _site/
```

```
make cockroachdb-build JEKYLLCONFIG=_config_prod_cockroachdb.yml
make cockroachcloud-build JEKYLLCONFIG=_config_prod_cockroachcloud.yml
htmltest -s
```

----

When running `make standatd` locally, it looks as if the stuff inside `_config_cockroachdb_local.yml` excludes a bunch of directories, as well as `_config_cockroachcloud.yml` is incompatible as a bunch of excludes leave to messiness. Would it instead help if I made a `make local-lint` command, which simulates the instructions above and make local linting work for you?

**Comment by rmloveland** [comment]
WOW that works for me!

After running everything as you describe above, I get:

```
htmltest -s
Skipping the checking of external links.
htmltest started at 03:00:01 on _site
========================================================================
running in concurrent mode, this is experimental
✔✔✔ passed in 3.047085634s
tested 2938 documents
```

@jseldess and @mjibson:

Can either of you think of any reason adding the targets & configs above would cause problems with our build process?  AFAICT it will be useful for running tests locally but should not affect anything else.

**Comment by rmloveland** [comment]
@mjibson and @otan, thank you both for your work on this.
